### DISCUSSION TITLE
Hardware Acceleration with amd HD4890

### DISCUSSION THREAD

**Discussion by hagar17** [blakeblackshear/frigate#8363] [General]
Hi All,
I have looked all over the place and tried my time to get my amd HD4890 into Frigate.

I have read the section in the doc 
"AMD/ATI GPUs (Radeon HD 2000 and newer GPUs) via libva-mesa-driver
VAAPI supports automatic profile selection so it will work automatically with both H.264 and H.265 streams.

Note: You also need to set LIBVA_DRIVER_NAME=radeonsi as an environment variable on the container.

ffmpeg:
  hwaccel_args: preset-vaapi"

I am running  SYSTEM 0.12.1-367D724 on
Home Assistant 2023.10.5
Supervisor 2023.10.1
Operating System 11.1
Frontend 20231005.0 - latest

If this question has already been answered please could you point me in the right direction
Thank you in advance for any help you may give

the question is where and how do I set the environment variable? is it in the home assistant config file or the frigate config .yaml
is there an example for the YAML that I need

"
mqtt:
  host: 192.168.0.xxx
  user: xxxxx
  password: xxxxxx

birdseye:
  # Optional: Enable birdseye view (default: shown below)
  enabled: false
  # Optional: Width of the output resolution (default: shown below)
  width: 640
    # Optional: Height of the output resolution (default: shown below)
  height: 480
    # Optional: Encoding quality of the mpeg1 feed (default: shown below)
    # 1 is the highest quality, and 31 is the lowest. Lower-quality feeds utilize less CPU resources.
  quality: 15
    # Optional: Mode of the view. Available options are: objects, motion, and continuous
    #   objects - cameras are included if they have had a tracked object within the last 30 seconds
    # Motion - cameras are included if motion was detected in the last 30 seconds
    #   continuous - all cameras are included always
  mode: motion

detectors:
  coral:
    type: edgetpu
    device: pci:0

go2rtc:
  streams:
    garden_reolink_camera: 
      - "ffmpeg:http://192.168.0.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=xxxx&password=xxxx#video=copy#audio=copy#audio=opus"
    garden_reolink_camera_sub: 
      - "ffmpeg:http://192.168.0.xxx/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=xxxxxx&password=xxxxx"
    door_reolink_camera: 
      - "ffmpeg:http://192.168.0.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=xxxxx&password=xxxxx#video=copy#audio=copy#audio=opus"
    door_reolink_camera_sub: 
      - "ffmpeg:http://192.168.0.xxx/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=xxxxx&password=xxxxx"
    lounge_reolink_camera: 
      - "ffmpeg:http://192.168.0.xxx/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=xxxxx&password=xxxxxx#video=copy#audio=copy#audio=opus"
    lounge_reolink_camera_sub: 
      - "ffmpeg:http://192.168.0.114/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=xxxxx&password=xxxxxx"

cameras:
  garden_reolink:
    ffmpeg:
      hwaccel_args: preset-vaapi
      inputs:
        - path: rtsp://127.0.0.1:8554/garden_reolink_camera?video=copy&audio=aac
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/garden_reolink_camera_sub?video=copy
          input_args: preset-rtsp-restream
          roles:
            - detect
    detect:
        # Optional: width of the frame for the input with the detect role (default: shown below)
      width: 640
        # Optional: height of the frame for the input with the detect role (default: shown below)
      height: 480
      fps: 5
      enabled: True
  # Optional: Number of frames without a detection before Frigate considers an object to be gone. (default: 5x the frame rate)
      max_disappeared: 25
    objects:
      track:
        - person
        - dog
        - car
    motion:
      mask:
        - 362,106,640,239,640,0,130,0,40,0,0,0,0,0,0,52
    record:
      enabled: True
      retain: 
        days: 2 
        #mode: active_objects 
      events:
        retain:
          default: 5 
          mode: motion 
        pre_capture: 3 
        post_capture: 10  
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 1
  door_reolink:
    ffmpeg:
        hwaccel_args: preset-vaapi         
        inputs:        
          - path: rtsp://127.0.0.1:8554/door_reolink_camera?video=copy&audio=aac
            input_args: preset-rtsp-restream
            roles:
              - record
          - path: rtsp://127.0.0.1:8554/door_reolink_camera_sub?video=copy
            input_args: preset-rtsp-restream
            roles:
              - detect
    detect:
        # Optional: width of the frame for the input with the detect role (default: shown below)
      width: 640
        # Optional: height of the frame for the input with the detect role (default: shown below)
      height: 480
      fps: 5
      enabled: True
  # Optional: Number of frames without a detection before Frigate considers an object to be gone. (default: 5x the frame rate)
      max_disappeared: 25
    objects:
      track:
        - person
        - dog
    record:
      enabled: true
      retain:
        days: 1
        mode: motion
      events:
        retain: 
          default: 2
          mode: motion
        pre_capture: 5
        post_capture: 10
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 1
  lounge_reolink:
    ffmpeg:
        hwaccel_args: preset-vaapi    
        inputs:        
          - path: rtsp://127.0.0.1:8554/lounge_reolink_camera?video=copy&audio=aac
            input_args: preset-rtsp-restream
            roles:
              - record
          - path: rtsp://127.0.0.1:8554/lounge_reolink_camera_sub?video=copy
            input_args: preset-rtsp-restream
            roles:
              - detect
    detect:
        # Optional: width of the frame for the input with the detect role (default: shown below)
      width: 640
        # Optional: height of the frame for the input with the detect role (default: shown below)
      height: 480
      fps: 5
      enabled: true
  # Optional: Number of frames without a detection before Frigate considers an object to be gone. (default: 5x the frame rate)
      max_disappeared: 25
    objects:
      track:
        - person
    record:
      enabled: true
      events:
        retain: 
          default: 2
          mode: motion
        pre_capture: 5
        post_capture: 10
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 1

If there is any other information you need to help please ask

**Comment by NickM-27** [comment]
Like the docs say, HA OS doesn't include the AMD drivers and can't be used

**Comment by hagar17** [comment]
Hi Nick, thank you for pointing that out I must have missed that part in the docs. I am clearly not the smartest guy here by a long way. could I ask a follow-up question? is it possible for me to add the drivers myself? or is it just better to go to the docker root for Frigate installation?

once again thank you for your help

**Reply by NickM-27** [reply]
to be clear this is called out here https://docs.frigate.video/frigate/installation#home-assistant-addon

this is not possible in HA OS because it is not designed to be adjusted, if that is the desire then frigate should be run in docker

**Reply by hagar17** [reply]
Hi NickM-27,
Thanks again for the help last week.
So I now have frigate install on Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-88-generic x86_64)
installed by docker-compose.yml see below:

the system ran fine until i un hash the line in the frigate config and it got no camera feed 

"#environment_vars:
  #LIBVA_DRIVER_NAME: radeonsi"

then i wont get any camera feed 

[log from frigate.txt](https://github.com/blakeblackshear/frigate/files/13258673/log.from.frigate.txt)
I have uploaded the log file 

any help in what I am doing wrong this time would be much appreciated

"#environment_vars:
  #LIBVA_DRIVER_NAME: radeonsi"

then i dont get any camera feed 

frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:stable
    shm_size: "64mb" # update for your cameras based on calculation above
    devices:
 #    - /dev/bus/usb:/dev/bus/usb # passes the USB Coral, needs to be modified for other versions
      - /dev/apex_0:/dev/apex_0 # passes a PCIe Coral, follow driver instructions here https://coral.ai/docs/m2/get-started/#2a-on-linux
      - /dev/dri/renderD128 # for intel hwaccel, needs to be updated for your hardware
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /opt/frigate/config/config.yaml:/config/config.yaml
      - /opt/frigate/media/storage:/media/frigate
      - type: tmpfs # Optional: 1GB of memory, reduces SSD/SD Card wear
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"
      - "8554:8554" # RTSP feeds
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
    environment:
      FRIGATE_RTSP_PASSWORD:








frigate config 
#yaml-language-server: $schema=http://192.168.0.163:5000/api/config/schema.json

mqtt:
  host: 192.168.0.202
  user: mqtt-user
  password:*********

detectors:
  coral:
    type: edgetpu
    device: pci:0

#environment_vars:
  #LIBVA_DRIVER_NAME: radeonsi

ffmpeg:
  hwaccel_args: preset-vaapi

birdseye:
  # Optional: Enable birdseye view (default: shown below)
  enabled: false
  # Optional: Width of the output resolution (default: shown below)
  width: 640
    # Optional: Height of the output resolution (default: shown below)
  height: 480
    # Optional: Encoding quality of the mpeg1 feed (default: shown below)
    # 1 is the highest quality, and 31 is the lowest. Lower quality feeds utilize less CPU resources.
  quality: 15
    # Optional: Mode of the view. Available options are: objects, motion, and continuous
    #   objects - cameras are included if they have had a tracked object within the last 30 seconds
    #   motion - cameras are included if motion was detected in the last 30 seconds
    #   continuous - all cameras are included always
  mode: motion

motion:
      # Optional: The threshold passed to cv2.threshold to determine if a pixel is different enough to be counted as motion. (default: shown below)
      # Increasing this value will make motion detection less sensitive and decreasing it will make motion detection more sensitive.
      # The value should be between 1 and 255.
  threshold: 15
      # Optional: Minimum size in pixels in the resized motion image that counts as motion (default: 30)
      # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
      # make motion detection more sensitive to smaller moving objects.
      # As a rule of thumb:
      #  - 15 - high sensitivity
      #  - 30 - medium sensitivity
      #  - 50 - low sensitivity
  contour_area: 15
      # Optional: Alpha value passed to cv2.accumulateWeighted when averaging the motion delta across multiple frames (default: shown below)
      # Higher values mean the current frame impacts the delta a lot, and a single raindrop may register as motion.
      # Too low and a fast moving person wont be detected as motion.
  delta_alpha: 0.2
      # Optional: Alpha value passed to cv2.accumulateWeighted when averaging frames to determine the background (default: shown below)
      # Higher values mean the current frame impacts the average a lot, and a new object will be averaged into the background faster.
      # Low values will cause things like moving shadows to be detected as motion for longer.
      # https://www.geeksforgeeks.org/background-subtraction-in-an-image-using-concept-of-running-average/
  frame_alpha: 0.2
      # Optional: Height of the resized motion frame  (default: 50)
      # This operates as an efficient blur alternative. Higher values will result in more granular motion detection at the expense
      # of higher CPU usage. Lower values result in less CPU, but small changes may not register as motion.
  frame_height: 50

go2rtc:
  streams:
    garden_reolink_camera:
      - "ffmpeg:http://192.168.0.1**/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=*****&password=*****#video=copy#audio=copy#audio=opus"
    garden_reolink_camera_sub:
      - "ffmpeg:http://192.168.0.1**/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=*******&password=*******

cameras:
  garden_reolink:
    ffmpeg:
      inputs:
          - path: rtsp://127.0.0.1:8554/garden_reolink_camera?video=copy&audio=aac
            input_args: preset-rtsp-restream
            roles:
              - record
          - path: rtsp://127.0.0.1:8554/garden_reolink_camera_sub?video=copy
            input_args: preset-rtsp-restream
            roles:
              - detect
    detect:
        # Optional: width of the frame for the input with the detect role (default: shown below)
      width: 896
        # Optional: height of the frame for the input with the detect role (default: shown below)
      height: 672
      fps: 7
      enabled: True
  # Optional: Number of frames without a detection before Frigate considers an object to be gone. (default: 5x the frame rate)
      max_disappeared: 25
    objects:
      track:
        - person
        - dog
        - car
        - bicycle
      filters:
        person:
          min_area: 9000
          max_area: 100000
          threshold: 0.75
    motion:
      mask:
        - 362,106,640,239,640,0,130,0,40,0,0,0,0,0,0,52
    record:
      enabled: True
      retain:
        days: 2
        #mode: active_objects 
      events:
        retain:
          default: 5
          mode: motion
        pre_capture: 3
        post_capture: 10
    snapshots:
      enabled: true
      timestamp: true
      bounding_box: true
      retain:
        default: 1
